plugin/trino-iceberg/src/main/java/io/trino/plugin/iceberg/catalog/file/FileMetastoreTableOperations.java

```diff
@@ -76,7 +80,13 @@ protected void commitToExistingTable(TableMetadata base, TableMetadata metadata)
             metastore.replaceTable(database, tableName, table, privileges);
         }
         catch (RuntimeException e) {
-            throw new CommitFailedException(e, "Failed to commit transaction to FileHiveMetastore");
+            if (e instanceof TrinoException trinoException &&
+                    trinoException.getErrorCode() == HIVE_METASTORE_ERROR.toErrorCode() &&
+                    Objects.equals(trinoException.getMessage(), "Cannot update Iceberg table: supplied previous location does not match current location")) {
```

Comment:
Basing the logic on exact message doesn't feel robust.
Maybe we need to have a dedicated error code for this.

Reply:
Thanks for the hint.
`Hive.HIVE_CONCURRENT_MODIFICATION_DETECTED` (already exists) seems appropriate for the purpose at hand.